rust/main/agents/relayer/src/relayer.rs

```diff
@@ -1068,11 +1099,113 @@ impl Relayer {
             .build_application_operation_verifiers(settings.origin_chains.iter(), core_metrics)
             .await
             .into_iter()
-            .filter_map(|(origin, app_context_verifier_res)| match app_context_verifier_res {
-                Ok(app_context_verifier) => Some((origin, app_context_verifier)),
+            .filter_map(
+                |(origin, app_context_verifier_res)| match app_context_verifier_res {
+                    Ok(app_context_verifier) => Some((origin, app_context_verifier)),
+                    Err(err) => {
+                        Self::record_critical_error(
+                            &origin,
+                            chain_metrics,
+                            &err,
+                            "Critical error when building application operation verifier",
+                        );
+                        None
+                    }
+                },
+            )
+            .collect()
+    }
+
+    pub async fn build_message_contract_syncs(
```

Comment:
these 3 functions are the exact same but just with different return types and a different critical error message right?

can we just make this 1 function with a generic return type and make the message be `"Critical error when building {data_type} contract sync"`?

Reply:
Done